### PR TITLE
Cover the functionalities of SIMD with more test cases

### PR DESCRIPTION
**Pull request by JesseCodeBones**
From: [2419](https://github.com/AssemblyScript/assemblyscript/issues/2419)

please kindly review if I missed any thing or if the test cases are suitable, thank you.

- [x] I've read the contributing guidelines
- [x] I've added my name and email to the NOTICE file

### CODE REVIEW

File: tests/compiler/simd.ts [hunk 1]
```diff
@@ -188,6 +188,91 @@ function test_v128(): void {
...
+    assert(v128.extract_lane<f32>(v, 2) == 71.0);
+    assert(v128.extract_lane<f32>(v, 3) == 4.0);
+  }
+  {
+    let v: v128 =v128.floor<f32>(f32x4(1.1, -0.25, 70.01, 4.0));
+    assert(v128.extract_lane<f32>(v, 0) == 1.0);
+    assert(v128.extract_lane<f32>(v, 1) == -1);
+    assert(v128.extract_lane<f32>(v, 2) == 70.0);
+    assert(v128.extract_lane<f32>(v, 3) == 4.0);
+  }
```
Comment (MaxGraey) [inline tests/compiler/simd.ts:220]: please use `== f32x4(1, -1, 70, 4)` instead extract_lane

File: tests/compiler/simd.ts [hunk 2]
```diff
@@ -188,6 +188,91 @@ function test_v128(): void {
...
+  {
+    let v: v128 = v128.ceil<f32>(f32x4(1.1, -0.25, 70.01, 4.0));
+    assert(v128.extract_lane<f32>(v, 0) == 2.0);
+    assert(v128.extract_lane<f32>(v, 1) == 0.0);
+    assert(v128.extract_lane<f32>(v, 2) == 71.0);
+    assert(v128.extract_lane<f32>(v, 3) == 4.0);
+  }
+  {
+    let v: v128 =v128.floor<f32>(f32x4(1.1, -0.25, 70.01, 4.0));
```
Comment (MaxGraey) [inline tests/compiler/simd.ts:215]:
```suggestion
    let v: v128 = v128.floor<f32>(f32x4(1.1, -0.25, 70.01, 4.0));
```

File: tests/compiler/simd.ts [hunk 3]
```diff
@@ -188,6 +188,91 @@ function test_v128(): void {
...
+        ,u8.MAX_VALUE
+        ,u8.MAX_VALUE
+        ,u8.MAX_VALUE
+        ,u8.MAX_VALUE
+        ,u8.MAX_VALUE
+        ,u8.MAX_VALUE
+        ,u8.MAX_VALUE)) 
+        == 
+        0x0000FFFF
+    );
```
Comment (MaxGraey) [inline tests/compiler/simd.ts:241]:
```suggestion
      v128.bitmask<i8>(i8x16(
        u8.MAX_VALUE,
        u8.MAX_VALUE,
        u8.MAX_VALUE,
        u8.MAX_VALUE,
        u8.MAX_VALUE,
        u8.MAX_VALUE,
        u8.MAX_VALUE,
        u8.MAX_VALUE,
        u8.MAX_VALUE,
        u8.MAX_VALUE,
        u8.MAX_VALUE,
        u8.MAX_VALUE,
        u8.MAX_VALUE,
        u8.MAX_VALUE,
        u8.MAX_VALUE,
        u8.MAX_VALUE
      )) == 0x0000FFFF
   );
```

File: tests/compiler/simd.ts [hunk 4]
```diff
@@ -188,6 +188,91 @@ function test_v128(): void {
...
+      v128.bitmask<i16>(i16x8(u16.MAX_VALUE, 
+        u16.MAX_VALUE, 
+        u16.MAX_VALUE, 
+        u16.MAX_VALUE, 
+        u16.MAX_VALUE, 
+        u16.MAX_VALUE, 
+        u16.MAX_VALUE, 
+        u16.MAX_VALUE))
+        == 
+        0x000000FF
```
Comment (MaxGraey) [inline tests/compiler/simd.ts:255]:
```suggestion
        u16.MAX_VALUE, 
        u16.MAX_VALUE, 
        u16.MAX_VALUE, 
        u16.MAX_VALUE, 
        u16.MAX_VALUE, 
        u16.MAX_VALUE, 
        u16.MAX_VALUE
     )) ==  0x000000FF
```

File: tests/compiler/simd.ts [hunk 5]
```diff
@@ -328,6 +413,55 @@ function test_i8x16(): void {
...
+      ,u8.MAX_VALUE
+      ,u8.MAX_VALUE
+      ,u8.MAX_VALUE
+      ,u8.MAX_VALUE
+      ,u8.MAX_VALUE
+      ,u8.MAX_VALUE
+      ,u8.MAX_VALUE)) 
+      == 
+      0x0000FFFF
+  );
```
Comment (MaxGraey) [inline tests/compiler/simd.ts:461]: same here

File: tests/compiler/simd.ts (hunk 1 above)
Comment (JesseCodeBones) [reply tests/compiler/simd.ts:220]: @MaxGraey 
for the function `floor`, there is no issue, but for `ceil`, it is not equal to `f32x4(2, 0, 71, 4)`.
here is my test code:  
```
export function test():bool {
    let v: v128 = f32x4.ceil(f32x4(1.1, -0.25, 70.01, 4.0));
    return (v == f32x4(2, 0, 71, 4));
}
```

Comment (MaxGraey) [reply tests/compiler/simd.ts]: Perhaps it should be `f32x4(2, -0.0, 71, 4)` ?

Comment (JesseCodeBones) [reply tests/compiler/simd.ts]: Yes, it is -0.0

File: tests/compiler/simd.ts [hunk 6]
```diff
@@ -188,6 +192,82 @@ function test_v128(): void {
...
+    v = v128.load64_lane(ptr, v, 0);
+    assert(v == i64x2(42, 0));
+    __free(ptr);
+  }
+  {
+    let v: v128 = v128.ceil<f32>(f32x4(1.1, -0.25, 70.01, 4.0));
+    assert(v == f32x4(2, -0.0, 71, 4));
+  }
+  {
+    let v: v128 =v128.floor<f32>(f32x4(1.1, -0.25, 70.01, 4.0))
```
Comment (MaxGraey) [inline tests/compiler/simd.ts:216]:
```suggestion
    let v: v128 = v128.floor<f32>(f32x4(1.1, -0.25, 70.01, 4.0))
```

**Review by MaxGraey — APPROVED** [review]

**Comment by MaxGraey** [comment]
Thanks!

**Pull request merged**